### DISCUSSION TITLE
I may have managed to narrow down the lagging problem in 2.2.9

### DISCUSSION THREAD

**Discussion by dannyshin76** [Budibase/budibase#9159] [Help]
[apm_update_test-export-1671754623714.tar.gz](https://github.com/Budibase/budibase/files/10291085/apm_update_test-export-1671754623714.tar.gz)

This is just 1 page excerpt (/login page) from my app that shows an unexpected behavior in 2.2.9 (works fine in 2.1.24)

- When the page loads up, it automatically loads preference data from BB table named "**theme**" into State variables (lang, apidomain, and passwordTTL). A BB table named "**system**" has the value to choose which theme to apply. For automatic setting for State variables, I used an embed script.
![image](https://user-images.githubusercontent.com/93232582/209247962-3790c41d-f2c1-4733-93da-7fbfa8873d1a.png)

- In 2.1.14, the page sets correct theme into State variables according to the theme.
- In 2.2.9, the page sets State values incorrectly. (I have modified the "Sign In" button to set the State variables again. It sets the values correctly)

(I suspect the cause may be some subtle difference when actions get executed between BB versions, which can surely lead to all kinds of backward compatibility failure.)

I hope this will help to solve the issue, because I really hope to be able to utilize all the great features of new BB version.

**Comment by melohagan** [comment]
Hey @dannyshin76 

This may be related to a quirk introduced with the skeleton loader:
- https://github.com/Budibase/budibase/discussions/9069
- https://github.com/Budibase/budibase/issues/9104

**Comment by shogunpurple** [comment]
I think we may have fixed this one on `develop` - I can't reproduce the errors when I import the above app into the latest `develop`.
